src/components/UpdatingForm/UpdatingForm.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+type Props = {
+  title: string;
+  id: number;
+  deleteTodo: (todoId: number) => void;
+  setIsUpdating: (isUpdating: boolean) => void;
+  onUpdateTodo: (todoId: number, netTitle: string) => void;
+};
+
+export const UpdatingForm: React.FC<Props> = ({
+  title,
+  id,
+  deleteTodo,
+  setIsUpdating,
+  onUpdateTodo,
+}) => {
+  const [newTitle, setNewTitle] = useState(title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, []);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+  };
+
+  const cancelEditing = () => {
+    setIsUpdating(false);
+  };
+
+  const submit = () => {
+    const trimmedNewTitle = newTitle.trim();
+
+    if (trimmedNewTitle === title) {
+      cancelEditing();
+
+      return;
+    }
+
+    if (!trimmedNewTitle) {
+      deleteTodo(id);
+
+      return;
+    }
+
+    onUpdateTodo(id, trimmedNewTitle);
+    setIsUpdating(false);
+  };
```

Comment:
The edit doesn't work

I add some words to the title.

![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/be781144-d0b5-42a0-9787-e483e612c050)

Click outside or press enter and the previous title comeback

![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/1ac7d812-b6a1-465a-b701-fddd1063d1a6)

It updates the title after the page is reloaded